janus_server/src/aggregator.rs

```diff
@@ -1935,63 +2099,1055 @@ mod tests {
+        let report_share_1 = generate_helper_report_share::<Prio3Aes128Count>(
+            task_id,
+            nonce_1,
+            hpke_recipient.config(),
+            &transcript_1.input_shares[1],
+        );
+
+        datastore
+            .run_tx(|tx| {
+                let (report_share_0, report_share_1) =
+                    (report_share_0.clone(), report_share_1.clone());
+                let (prep_step_0, prep_step_1) = (prep_step_0.clone(), prep_step_1.clone());
+
+                Box::pin(async move {
+                    tx.put_task(&TaskParameters::new_dummy(
+                        task_id,
+                        Vec::new(),
+                        Vdaf::Prio3Aes128Count,
+                        Role::Helper,
+                    ))
+                    .await?;
+
+                    tx.put_report_share(task_id, &report_share_0).await?;
+                    tx.put_report_share(task_id, &report_share_1).await?;
+
+                    tx.put_aggregation_job(&AggregationJob::<Prio3Aes128Count> {
+                        aggregation_job_id,
+                        task_id,
+                        aggregation_param: (),
+                        state: AggregationJobState::InProgress,
+                    })
+                    .await?;
+
+                    tx.put_report_aggregation(&ReportAggregation::<Prio3Aes128Count> {
+                        aggregation_job_id,
+                        task_id,
+                        nonce: nonce_0,
+                        ord: 0,
+                        state: ReportAggregationState::Waiting(prep_step_0),
+                    })
+                    .await?;
+                    tx.put_report_aggregation(&ReportAggregation::<Prio3Aes128Count> {
+                        aggregation_job_id,
+                        task_id,
+                        nonce: nonce_1,
+                        ord: 1,
+                        state: ReportAggregationState::Waiting(prep_step_1),
+                    })
+                    .await
+                })
+            })
+            .await
+            .unwrap();
+
+        let request = AggregateReq {
+            task_id,
+            job_id: aggregation_job_id,
+            body: AggregateContinueReq {
+                seq: vec![Transition {
+                    nonce: nonce_0,
+                    trans_data: TransitionTypeSpecificData::Continued {
+                        payload: prep_msg_0.get_encoded(),
+                    },
+                }],
+            },
+        };
+
+        // Create aggregator filter, send request, and parse response.
+        let filter = aggregator_filter(
+            vdaf,
+            datastore.clone(),
+            clock,
+            skew,
+            Role::Helper,
+            verify_params[1].clone(),
+            hpke_recipient,
+            hmac_key.clone(),
         )
+        .unwrap();
+
+        let mut response = warp::test::request()
```

Comment:
I disagree for a couple of reasons:
1. "Test the public interface." Arguably, the public interface is the `warp` filter. (or, going further, it might be the server itself, since it's `aggregator_server` that's marked `pub` -- but I'm OK viewing these as effectively identical.)

2. There actually is functionality under test that is part of the filter. Namely, authentication, request parsing, response encoding, and exposing the endpoint only if acting as a helper. I think it's useful/important to test these as well.


For the problem mentioned around #55, perhaps we should factor things so that "look up task parameters (including VDAF) from task ID" is a generic operation -- eventually we're going to want to be able to do this per-request, possibly falling back to a DB lookup for new tasks, so I think it makes sense to do so. Then tests could just use a lookup function that returns the desired parameters, including a `fake::Vdaf`.

Reply:
Yes, I have to admit that part of my approach in #55 was to limit the size of the changes given that the PR is already large. In any case, since I'm on vacation from tomorrow, we should favour your change over mine in the name of unblocking yours and David's work.